data-table/__tests__/table.js

```diff
@@ -108,10 +127,30 @@ describe('API basic tests', () => {
 });
 
 describe('Interaction tests', () => {
-    test('User can input to cell', () => {
-        table.createDataTable({
-            'wrapperDivId': 'div-id'
-        });
+    test('Invalid input to text input field updates cell appropriately', () => {
+        createSingleCellTable(['Value'], [Number], [0], [invalidIfNegative])
+        expect(document.getElementsByClassName('invalid-cell').length).toEqual(0);
+        expect(document.getElementsByClassName('data-table-cell').length).toEqual(4);
+
+        let input = document.getElementsByClassName('cell-input')[0];
+        input.value = -23;
+        input.dispatchEvent(new Event('focusout'));
+
+        expect(document.getElementsByClassName('invalid-cell').length).toEqual(1);
+        expect(document.getElementsByClassName('data-table-cell').length).toEqual(3);
```

Comment:
Why does `data-table-cell` decrease?

Consider adding "helper functions" which describe what's happening, since you'll be reusing these frequently throughout the tests:
`numErrorsVisible()` and `numCellsVisible()` would each be one line of code, but would make the `expect`s read nicely:
`expect(numErrorsVisible()).toEqual(1);`

(This would also be helpful if, for instance, you change the functionality so that `invalid-cell` always exists but is set to `style: display=none` - you'd only need to change the code in one place.)

Reply:
So right now I have the `invalid-cell` CSS replacing the `data-table-cell` CSS whenever there's an error so the number of elements with the `data-table-cell` decrease. Added a FIXME to that particular section of code so it doesn't get forgotten.

Added helper function.